src/common/io/src/buffer/buffer_read_datetime_ext.rs

```diff
@@ -97,15 +97,26 @@ where R: BufferRead
             loop {
                 buf.clear();
                 let size = self.keep_read(&mut buf, |f| (b'0'..=b'9').contains(&f))?;
-                match get_time(&mut buf, size) {
-                    Ok(time) => times.push(time),
-                    Err(e) => return Err(e),
-                }
-                if times.len() == 3 {
+                if size == 0 {
                     break;
+                } else {
+                    let time = get_time(&mut buf, size)?;
+                    times.push(time);
+                    if times.len() == 3 {
+                        break;
+                    }
+                    self.ignore_byte(b':')?;
                 }
-                self.must_ignore_byte(b':')?;
             }
+            // Time part is HH:MM or HH
```

Comment:
better to add some comments and examples to explain this special case.

Reply:
Done